docs/specifications/update.md

```diff
@@ -0,0 +1,191 @@
+| `--next` | `boolean` | If true, allows version discovery to include Beta and RC. Defaults to false. |
+| `--migration-only` | `boolean` | If true, don't change the `package.json` file, only apply migration scripts. |
+| `--from` | `version` | Apply migrations from a certain version number. |
+| `--to` | `version` | Apply migrations up to a certain version number (inclusive). By default will update to the tag selected. |
+
+## Details
+
+The schematic performs the following steps, in order:
+
+1. Get all installed package names and versions from the `package.json` into `dependencyMap: Map<string, string>`.
+1. From that map, fetch all `package.json` from the NPM repository, which contains all versions, and gather them in a `Map<string, NpmPackageJson>`.
+  1. At the same time, update the `Map<>` with the version of the package which is believed to be installed (largest version number matching the version range).
+  1. **WARNING**: this might not be the exact installed versions, unfortunately. We should have a proper `package-lock.json` loader, and support `yarn.lock` as well, but these are stretch goals (and where do we stop).
+1. For each packages mentioned on the command line, update to the target version (by default largest non-beta non-rc version):  
+
+  ```python
+  # ARGV    The packages being requested by the user.
+  # NPM     A map of package name to a map of version to PackageJson structure.
+  # V       A map of package name to available versions.
+  # PKG     A map of package name to PackageJson structure, for the installed versions.
+  # next    A flag for the "--next" command line argument.
+
+  # First add all updating packages' peer dependencies. This should be recursive but simplified
+  # here for readability.
+  ARGV += [ NPM[p][max([ v for v in V[p] if (not is_beta(v) or next) ])].peerDependencies
+            for p in ARGV ]
+
+  for p in ARGV:
+    x = max([ v for v in V[p] if (not is_beta(v) or next) ])
+
+    for other in set(PKG.keys()) - set([ p ]):
+      # Verify all packages' peer dependencies.
+      if has(other.peerDependencies, p) and !compatible(x, other.peerDependencies[p]):
+        showError('Cannot update dependency "%s": "%s" is incompatible with the updated dependency' % (x, other))
+
+      if any( has(other.peerDependencies, peer) and !compatible(x, other.peerDependencies[peer])
+              for peer in PKG[p].peerDependencies.keys() ):
+        showError('Cannot update dependency "%s": "%s" depends on an incompatible peer dependency' % (x, other))
+
+    update_package_json(p, x)
+```
+
+
+
+## Library Developers
+
+Libraries are responsible for defining their own update schematics. The `ng update` tool will update the package.json, and if it detects a special key in package.json of the library, will run the update schematic on it (with version information metadata).
```

Comment:
"special key" indicates some sort of secret here, either use it here or mention that it is described below.

Reply:
Clarified.